spec/services/courses/generate_course_title_service_spec.rb

```diff
@@ -0,0 +1,130 @@
+require "rails_helper"
+
+describe Courses::GenerateCourseTitleService do
+  let(:service) { described_class.new }
+  let(:subjects) { [] }
+  let(:is_send) { false }
+  let(:level) { "primary" }
+  let(:course) { Course.new(level: level, subjects: subjects, is_send: is_send) }
+  let(:modern_languages) { create(:subject, subject_name: "Modern Languages", type: :SecondarySubject).becomes(SecondarySubject) }
+  let(:generated_title) { service.execute(course: course) }
+
+  before { modern_languages }
+
+  shared_examples "with SEND" do
+    context "With SEND" do
+      let(:is_send) { true }
+
+      it "Appends SEND information to the title" do
+        expect(generated_title).to end_with("with Special educational needs and disability")
+      end
+    end
+  end
+
+  context "With no subjects" do
+    it "Returns an empty string" do
+      expect(generated_title).to eq("")
+    end
+  end
+
+  context "Generating a title for a further education course" do
+    let(:level) { "further_education" }
+
+    it "returns 'Further education'" do
+      expect(generated_title).to eq("Further education")
+    end
+
+    include_examples "with SEND"
+  end
+
+  context "Generating a title for non-further education course" do
+    let(:level) { "primary" }
+
+    context "With a single subject" do
+      context "Example one" do
+        let(:subjects) { [Subject.new(subject_name: "Cats")] }
+        it "Returns the subject name" do
+          expect(generated_title).to eq("Cats")
+        end
+
+        include_examples "with SEND"
+      end
+
+      context "Example two" do
```

Comment:
Why do we need two examples?

Reply:
I was TDD-ing it out and use examples for triangulation. I usually remove these before committing for this codebase because no where else has it so I wanna stay consistent :eyes: i missed this one though